.github/workflows/release.yaml

```diff
@@ -21,6 +24,16 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
+      - name: Submodule Tag
+        if: ${{ github.event.inputs.submodule_tag }}
+        run: |
+          if ! [[ ${{ github.event.inputs.submodule_tag }} =~ [A-Za-z0-9\-\_]+\/{1}v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
```

Comment:
I'd probably would string split `submodule_tag` at `/`, e.g. extracting `dir=client` and `version=v1.2.3` from `client/v1.2.3`.

Then check if directory `client` exists (and maybe even if it contains a `go.mod` file). If not -> exit
 
Then use the same regex on `version` like in the next step.

If all succeeds `git tag  client/v1.2.3`. WDYT?